pylint/test/spelling/unittest_checker_spelling.py

```diff
@@ -12,44 +12,44 @@
 
 """Unittest for the spelling checker."""
 
+import subprocess
+import os
+
 import pytest
 
 import astroid
 
 from pylint.checkers import spelling
-from pylint.testutils import CheckerTestCase, Message, set_config, _tokenize_str
+from pylint.testutils import CheckerTestCase, Message, _tokenize_str
+
+
+@pytest.fixture
+def dict_name(request):
+    return request.config.getoption("--spelling-dict-name", skip=True)
 
-# try to create enchant dictionary
-try:
-    import enchant
-except ImportError:
-    enchant = None
 
-spell_dict = None
-if enchant is not None:
-    try:
-        enchant.Dict("en_US")
-        spell_dict = "en_US"
-    except enchant.DictNotFoundError:
-        pass
+@pytest.fixture
+def dict_paths(request):
+    return request.config.getoption("--spelling-dict-paths", skip=True)
 
 
 class TestSpellingChecker(CheckerTestCase):
     CHECKER_CLASS = spelling.SpellingChecker
 
-    skip_on_missing_package_or_dict = pytest.mark.skipif(
-        spell_dict is None,
-        reason="missing python-enchant package or missing spelling dictionaries",
-    )
-
     def _get_msg_suggestions(self, word, count=4):
-        return "'{}'".format(
-            "' or '".join(self.checker.spelling_dict.suggest(word)[:count])
+        return "Did you mean: '{}'?".format(
+            "' or '".join(self.checker.hobj.suggest(word)[:count])
         )
 
-    @skip_on_missing_package_or_dict
-    @set_config(spelling_dict=spell_dict)
-    def test_check_bad_coment(self):
+    def _config(self, **kwargs):
+        for key, value in kwargs.items():
+            setattr(self.checker.config, key, value)
+        self.checker.open()
+        if not self.checker.initialized:
+            pytest.skip("Missing hunspell package or missing spelling dictionaries")
+
+    def test_check_bad_coment(self, dict_name, dict_paths):
```

Comment:
Any reason why we can't use `set_config` here instead? It feels weird that we're reinventing what `set_config` does, I assume just to use the pytest fixtures.